### PR TITLE
Make dispatched events really final

### PR DESCRIPTION
**Pull request by Tobion**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Continuation of #33152 for master

### CODE REVIEW

**Review by ro0NL — COMMENTED** [review]
shouldnt we update tests in 4.4 also?

**Comment by Tobion** [comment]
Extracted updating tests for 4.4 in #33482

**Comment by nicolas-grekas** [comment]
rebase unlocked.

**Comment by Tobion** [comment]
Rebased and tests pass

File: src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php [hunk 1]
```diff
@@ -35,20 +33,16 @@ public function __construct(Command $command, InputInterface $input, OutputInter
 
     /**
      * Sets the exit code.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php]: let's remove that too :)

File: src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php [hunk 2]
```diff
@@ -35,20 +33,16 @@ public function __construct(Command $command, InputInterface $input, OutputInter
...
      */
-    public function setExitCode($exitCode)
+    public function setExitCode(int $exitCode): void
     {
-        $this->exitCode = (int) $exitCode;
+        $this->exitCode = $exitCode;
     }
 
     /**
      * Gets the exit code.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php]: drop drop drop :)

File: src/Symfony/Component/HttpKernel/Event/ControllerEvent.php [hunk 3]
```diff
@@ -40,15 +38,13 @@ public function __construct(HttpKernelInterface $kernel, callable $controller, R
 
     /**
      * Returns the current controller.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Event/ControllerEvent.php]: drop?

File: src/Symfony/Component/HttpKernel/Event/ExceptionEvent.php [hunk 4]
```diff
@@ -52,10 +50,8 @@ public function __construct(HttpKernelInterface $kernel, Request $request, int $
 
     /**
      * Returns the thrown exception.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpKernel/Event/ExceptionEvent.php]: drop?

File: src/Symfony/Component/Security/Http/Event/InteractiveLoginEvent.php [hunk 5]
```diff
@@ -33,20 +31,16 @@ public function __construct(Request $request, TokenInterface $authenticationToke
 
     /**
      * Gets the request.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Http/Event/InteractiveLoginEvent.php]: drop

File: src/Symfony/Component/Security/Http/Event/InteractiveLoginEvent.php [hunk 6]
```diff
@@ -33,20 +31,16 @@ public function __construct(Request $request, TokenInterface $authenticationToke
...
-     * @return Request A Request instance
      */
-    public function getRequest()
+    public function getRequest(): Request
     {
         return $this->request;
     }
 
     /**
      * Gets the authentication token.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Http/Event/InteractiveLoginEvent.php]: drop

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php (hunk 1 above)
Comment (Tobion) [reply src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php]: all those changes would be for 4.x as we also removed other irrelevant phpdocs there. so it shouldn't be part of this PR or?

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php]: for this pr it's fine to me, partial removals are less useful than full removals to me

Comment (Tobion) [reply src/Symfony/Component/Console/Event/ConsoleTerminateEvent.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Tobion.

**Pull request merged**